### PR TITLE
sample: fix sidewalk_thread msgq

### PR DESCRIPTION
**Pull request by RobertGalatNordic**
## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: master
```

## Description

JIRA ticket: KRKNWK-19262
by statically initializing the queue we prevent issues with events.

## Self review

- [x] There is no commented code.
- [x] There are no TODO/FIXME comments without associated issue ticket.
- [x] Commits are properly organized.
- [x] Change has been tested.
- [ ] <s>Tests were updated (if applicable).</s>

### CODE REVIEW

**Review by totyz — APPROVED** [review]
Test from the ticket pass without BUS FAULT

File: samples/sid_end_device/src/sidewalk_events.c [hunk 1]
```diff
@@ -326,6 +353,7 @@ void sidewalk_event_exit(sidewalk_ctx_t *sid, void *ctx)
 		list_element = sys_slist_get(&pending_message_list);
 	}
 	k_mutex_unlock(&pending_message_list_mutex);
+	sid->handle = NULL;
```
Comment (ktaborowski) [inline samples/sid_end_device/src/sidewalk_events.c:356]: Isn't it handled by sid api (init/deinit)?

**Review by ktaborowski — APPROVED** [review]
lgtm

File: samples/sid_end_device/src/sidewalk_events.c (hunk 1 above)
Comment (RobertGalatNordic) [reply samples/sid_end_device/src/sidewalk_events.c:356]: sid api can not NULL this pointer on its own.

**Pull request merged**